bin/cheribsdtest/cheribsdtest.c

```diff
@@ -431,6 +450,13 @@ cheribsdtest_run_test(const struct cheri_test *ctp)
 		    ccsp->ccs_si_trapno);
 		goto fail;
 	}
+	if ((ctp->ct_flags & CT_FLAG_SI_ADDR) && 
+	    cheri_pointers_equal(ccsp->ccs_si_addr_expected, ccsp->ccs_si_addr)) {
+		snprintf(reason, sizeof(reason), "Expected si_addr %#lp, got %#lp",
+		    (void * __capability) ccsp->ccs_si_addr_expected,
+		    (void * __capability) ccsp->ccs_si_addr);
```

Comment:
They're void * not void * __capability, so print them as such. Also, no spaces after casts.

Reply:
OK, in that case I need a format string `#define` for printing a pointer that will work in hybrid or purecap modes. Is there one already? If not where should I put it and what should I call it?

Reply:
Just drop the l's from %#lp